bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeConverter.java

```diff
@@ -606,33 +629,51 @@ private static boolean isConvertibleToTupleType(Object sourceValue, BTupleType t
         int initialErrorCount;
         String elementIndex;
         boolean returnVal = true;
+        Set<Type> convertibleTypes;
         for (int i = 0; i < targetTypeSize; i++) {
             initialErrorCount = errors.size();
             elementIndex = getElementIndex(varName, i);
-            if (!isConvertibleToArrayInstance(source.getRefValue(i), targetTypes.get(i), unresolvedValues,
-                    elementIndex, errors)) {
+            convertibleTypes = getConvertibleTypes(source.getRefValue(i), targetTypes.get(i), elementIndex,
+                    false, unresolvedValues, errors);
+            if (convertibleTypes.isEmpty()) {
                 addErrorMessage(errors.size() - initialErrorCount, errors, "tuple element '" +
                         elementIndex + "' should be of type '" + targetTypes.get(i).toString() + "', found '" +
                         getShortSourceValue(source.getRefValue(i)) + "'");
-                if (errors.size() >= MAX_CONVERSION_ERROR_COUNT + 1) {
-                    return false;
-                }
+                returnVal = false;
+            } else if (convertibleTypes.size() > 1 && !convertibleTypes.contains(TypeChecker.getType(
+                    source.getRefValue(i))) && !hasIntegerSubTypes(convertibleTypes)) {
+                addErrorMessage(errors.size() - initialErrorCount, errors,
+                        "tuple element '" + elementIndex + "' cannot be converted to '" +
+                                targetTypes.get(i) + "': ambiguous target type");
                 returnVal = false;
             }
+            if (errors.size() >= MAX_CONVERSION_ERROR_COUNT + 1) {
```

Comment:
Is this cover in the tests? There are a few more lines showed as not covered in tests

Reply:
Removed unnecessary checks and added a test case for it.